File: src/main/java/io/github/hadixlin/iss/KeepEnglishInNormalExtension.java
package io.github.hadixlin.iss;

import org.jetbrains.annotations.NotNull;

/**
 * Created by hadix on 09/04/2017.
 */
public class KeepEnglishInNormalExtension
        extends KeepEnglishInNormalAndRestoreInInsertExtension {

    public KeepEnglishInNormalExtension() {
        super(false);
    }

    @NotNull
    @Override
    public String getName() {
        return "keep-english-in-normal";
    }
}

File: src/main/kotlin/io/github/hadixlin/iss/KeepEnglishInNormalExtension.kt
package io.github.hadixlin.iss

import com.maddyhome.idea.vim.extension.VimExtension

/**
 * @author hadix
 * @date 09/04/2017
 */
class KeepEnglishInNormalExtension : VimExtension {

    override fun getName(): String {
        return NAME
    }

    override fun init() {
        InputMethodAutoSwitcher.enable()
    }


    override fun dispose() {
        InputMethodAutoSwitcher.disable()
    }

    companion object {
        const val NAME = "keep-english-in-normal"
    }

}
